### ISSUE TITLE
Choosing a new color for star ratings in the Plugins Directory and Forums

### ISSUE THREAD

**Issue by ndiego** [WordPress/wordpress.org#253]
I am creating this issue to provide a more central location for discussion about the star color for reviews in the Plugins Directory and Forums. The discussion originated in [WP Slack](https://wordpress.slack.com/archives/C04U953K77A/p1712738229057149?thread_ts=1712717222.251569&cid=C04U953K77A). 

Recently, the [Forums were refreshed](https://make.wordpress.org/meta/2024/04/10/the-forums-get-a-refresh/), and the star rating color for plugins/themes was updated, first to black and then to pomegranate `#E26F56`, which is [one of the secondary brand colors](https://www.figma.com/file/HOJTpCFfa3tR0EccUlu0CM/WordPress.org-Design-Library?type=design&node-id=101%3A4217&mode=design&t=fAVhWTigElfqr9Mm-1) for WordPress.org. **The color was changed because the original yellow color `#ffc733` did not meet the accessibility requirements ([3:1 UI contrast](https://contrast.tools?tab=wcag-21&text=e26f56&background=ffffff)).**

| Forums (current design) | Forums (previous design) |
|-|-|
|<img width="1439" alt="image" src="https://github.com/WordPress/wordpress.org/assets/4832319/dfc81086-24ef-40c9-936b-1da287d630f3">|<img width="1438" alt="image" src="https://github.com/WordPress/wordpress.org/assets/4832319/28ebec80-8330-470d-b484-1e41c0c06967">|

A number of colors have been proposed. Here are a few: 
| Star alternatives | Star alternatives |
|-|-|
|![preferred](https://github.com/WordPress/wordpress.org/assets/4832319/6eafc807-e801-4a6e-8408-7c4eeb710027)|![tests (1)](https://github.com/WordPress/wordpress.org/assets/4832319/8154f071-83e1-4e91-b36d-73478d7eb2c0)|

You can also check out this [Figma file for more options](https://www.figma.com/file/pRdW39E6SE3q8S6e7558cl/WP.org-Rating-Stars?type=design&node-id=0-1&mode=design). 

The pomegranate color is currently implemented both in [Forums](https://wordpress.org/support/plugin/gutenberg/reviews/?old-theme=0) and in the demo site for the [Plugins Directory](https://test.wordpress.org/plugins/gutenberg/) refresh, so you can test this live. Share your thoughts on this issue.

**Comment by courtneyr-dev** [comment]
Is there a Needs Accessibility label?

**Comment by joedolson** [comment]
For accessibility, any of the proposed options are viable. The potential problem with the outlined option would be insufficient difference between the filled star and the empty star, but omitting the border on the empty resolves that. 

For style, I prefer the yellow with an outline.

**Comment by zackkatz** [comment]
~⭐️ I vote yellow star with outline! 🗳️~

EDIT: I vote pomegranate due to the comment quoted below (FYI, at the time of editing, my previous comment had 3 thumbs-up emoji votes)

> Would the color we pick also apply to the rating bars? I'm not sure that yellow-with-border would work well here.
> 
> <img alt="Screenshot 2024-04-15 at 4 50 31 PM" width="374" src="https://private-user-images.githubusercontent.com/541093/322625720-c65552d0-bbae-42a1-a7b7-309ed38d9eea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyMTcxNTEsIm5iZiI6MTcxMzIxNjg1MSwicGF0aCI6Ii81NDEwOTMvMzIyNjI1NzIwLWM2NTU1MmQwLWJiYWUtNDJhMS1hN2I3LTMwOWVkMzhkOWVlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNVQyMTM0MTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZmUyOWU4Y2M4YWE1ZTM5ODEzNmI4NzlmNDRlYzA4ZTdmODVjYjM4MmIxM2IzZjViYzY1NmEzODFhMmY5YzM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.oOyTDDMO_RSbd3D3OmpUvtX4JT1LtQCS7YPhDFST2q4">
> And to share my opinion— I like the pomegranate stars 🙂

**Comment by jazzsequence** [comment]
> The potential problem with the outlined option would be insufficient difference between the filled star and the empty star, but omitting the border on the empty resolves that.

This was my first thought as well. +1 yellow star w/outline (1st) or pomegranate (2nd).

**Comment by ndiego** [comment]
While I agree that the yellow star with the border looks nice, there are some dev considerations. Implementing this option will not be as straightforward as switching out colors. Coupled with the fact that pomegranate is a secondary brand color, my vote is for pomegranate.

**Comment by ryelle** [comment]
Would the color we pick also apply to the rating bars? I'm not sure that yellow-with-border would work well here.

<img width="374" alt="Screenshot 2024-04-15 at 4 50 31 PM" src="https://github.com/WordPress/wordpress.org/assets/541093/c65552d0-bbae-42a1-a7b7-309ed38d9eea">

And to share my opinion— I like the pomegranate stars 🙂

**Comment by ndiego** [comment]
> Would the color we pick also apply to the rating bars? I'm not sure that yellow-with-border would work well here.

That's a great point that I completely forgot about. 😅

**Comment by codente** [comment]
+1 for pomegranate

**Comment by AnthonyLedesma** [comment]
I love that pomegranate `#E26F56` color.

**Comment by ghost** [comment]
I use dark mode. Things like this tend to not show up. Can you show the examples in dark mode?

**Comment by jasmussen** [comment]
Pomegranate works well in that it's a flat color, which vibes with most other iconography, yet it passes contrast, and it's part of [the website brand](https://www.figma.com/file/HOJTpCFfa3tR0EccUlu0CM/WordPress.org-Design-Library?type=design&node-id=101%3A4217&mode=design&t=BXdAWlVIn6lPTI08-1). 

@lada7042

> I use dark mode. Things like this tend to not show up. Can you show the examples in dark mode?

The website doesn't yet have official dark mode built-in, though I'd like to see that happen at some point in the future. [Openverse](https://openverse.org) has incoming dark mode support, I would expect that wp.org could follow the same model. There are [designs for Openverse dark mode here](https://make.wordpress.org/design/2024/03/25/design-share-mar-11-mar-22/).

That said, the pomegranate works well on dark backgrounds, here's an example:

<img width="311" alt="emulated dark mode" src="https://github.com/WordPress/wordpress.org/assets/1204802/49a7fe5f-2e4d-45d0-be3f-944991fc5c83">

As noted, on white the color exceeds the minimum 3:1 contrast by achieving 3.16:1. On the dark background it surpasses that with a comfortable 4.7:1 contrast ratio. Tested using [Contrast Tools](https://contrast.tools/?tab=wcag-21).

[Figma](https://www.figma.com/file/pRdW39E6SE3q8S6e7558cl/WP.org-Rating-Stars?type=design&node-id=0%3A1&mode=design&t=HUNj6tAmWrDQb9Zu-1), for anyone who wants to tinker.

**Comment by courtneyr-dev** [comment]
> For accessibility, any of the proposed options are viable. The potential problem with the outlined option would be insufficient difference between the filled star and the empty star, but omitting the border on the empty resolves that.
> 
> For style, I prefer the yellow with an outline.

@joedolson  How do you feel about the images with a 1/2 star? Border or no border on the empty half? 

I'm thinking no border to keep things more simple visually.

**Comment by courtneyr-dev** [comment]
I inquired about the various color options with a friend who cannot see gradients of color, only "blue" or "orange", not shades of these.  The friend's preference was for the pomegranate as well.

In Joen's last reply, the pomegranate strikes me as nearly orange, not feeling red. This is good in the context of the forums. Red could connotate frustration for at least many English speaking locales. I don't have experience with emotional representation in colors in other regions.

**Comment by ndiego** [comment]
> In Joen's last reply, the pomegranate strikes me as nearly orange, not feeling red.

Yeah, to my eyes, pomegranate is closer to orange than red. 

- `#FF0000` - Red
- `#E26F56` - Pomegranate
- `#ffa500` - Orange
- `#ffc733` - Old yellow

ChatGPT 4 thinks so as well 😂

> The color represented by the hex code #E26F56 is closer to orange. This hex code produces a shade that is often described as a soft or pastel red-orange, leaning more towards orange due to its warm and bright characteristics.

**Comment by mujuonly** [comment]
+1 Yellow

**Comment by ndiego** [comment]
Given the feedback above, I propose we move forward with `#E26F56` so as not to block the Plugin Directory refresh launch but keep this discussion open for a potential future update. Please continue to share your feedback here 🙏

cc @WordPress/meta-design @StevenDufresne @adamwoodnz

**Comment by NekoJonez** [comment]
Why not give the users the choice out of the options and let us pick a default here?

**Comment by amberhinds** [comment]
I like the pomegranate but have some concerns. Technically, if we're always outputting the text (4.5 out of 5 stars) then these are decorative and don't have to pass anything and can be hidden from screen readers. 

When we have scenarios like this, though, where sighted people have no text equivalent, they should pass AA at a minimum.
<img width="390" alt="Screenshot 2024-04-16of reviews heading and stars with no text descriptor" src="https://github.com/WordPress/wordpress.org/assets/13153456/2792f971-3eea-40bb-b5e1-5be74dff424a">

 Arguably, in this instance, they don't count as a UI component but as something meant to function as a text equivalent. I.e., the average star rating of a plugin is really important for many people when selecting plugins. If we consider the importance of these stars, the pomegranate fails on white because it's not at least 4.5:1. 

IMO, an improvement here, if you didn't want to have to make the stars darker, would be to have the stars (in any color) and text in a color that passes 7:1 (AAA) to the right of the stars.

**Comment by unscripted** [comment]
+1 for pomegranate.

I also really like the idea of outputting the rating in text. While not within the scope of this issue, in addition to accessibility, it could also improve schema markup.

**Comment by joedolson** [comment]
Having the rating in text would be great. It used to be available via a title attribute, which was removed because it's not an accessible way to communicate information, but it should have been added in text at the same time. See https://meta.trac.wordpress.org/changeset/3792

**Comment by jasmussen** [comment]
> If we consider the importance of these stars, the pomegranate fails on white because it's not at least 4.5:1.

As UI elements that are relatively large (equivalent to 18pt text) I'd love to understand why these would need to meet 4.5:1 text contrast vs. 3:1 UI/large text contrast. 

That said, I'd also like to see adjacent text next to every instance of the stars. It could be as simple as ★★★☆☆ (3/5).

**Comment by joedolson** [comment]
They aren't large text; they're 20px. Large text is equivalent to 18pt, which is approximately 24px. Since they're icons, it's kind of hard to make that equivalency - they have completely different concepts of line and fill than a text font would. 

I don't think WCAG provides any real guidance on how to assess the color contrast of icons used to communicate information; by that standard, this is more equivalent to an image like a graph or chart, and alt text would generally be considered sufficient there.   

Honestly, it would be *much* better to ignore the whole question of whether and how WCAG covers this particular scenario and just have the rating visible in plain text.

**Comment by flexseth** [comment]
Another thing to consider may be new changes that come to the WordPress Admin - which colors will work best based on the default color scheme there?

**Comment by mathetos** [comment]
The use of the orange/pomegrade color for Stars is probably a bit subjective, but often red-ish colors in the context of ratings/reviews is perceived as a negative, not a positive. 

Current color:
<img width="757" alt="image" src="https://github.com/WordPress/wordpress.org/assets/930724/a680fc27-74c8-4fbc-b288-10c84ce1a9cc">

Also, looking through the whole new homepage and rest of the site, there is no use of an accent color at all, it's all just shades of blue, grey, and black.

But I can see that using blue or black for the stars is also not sufficient. 

But in reviewing the color pallete in the stylesheet, I see a reference to a nice green:
#00d084 `--wp--preset--color--vivid-green-cyan` 

Here's what that looks like on the Plugins page:
<img width="749" alt="image" src="https://github.com/WordPress/wordpress.org/assets/930724/83552888-a2c1-474b-93b4-47a3e8e8b7ca">

Here's what it looks like on the single plugin page:
<img width="789" alt="image" src="https://github.com/WordPress/wordpress.org/assets/930724/23396963-db5b-4f11-95b1-cbf079fb870f">


Again, color schemes can be subjective for sure, but the way in which the current orange/red evokes a negative connotation I don't find to be ideal, or the best color choice in the context of the rest of the website. The green I feel is more of a natural fit both for connotation and in the broader context of the whole website.

**Comment by Abhiramrg** [comment]
Color schemes are certainly subjective, but I don't think the current orange/Pomegranate hue gives off the best impression. Its association with negativity doesn't seem ideal, especially considering the overall aesthetic of the website. Personally, I find incorporating green for the rating icons and bars would be more suitable, both in terms of connotation and its compatibility with the website's theme.

**Comment by ndiego** [comment]
I am going to close out this issue since it's been a couple of months, and the pomegranate color has been deployed on [Plugins](https://wordpress.org/plugins/) and [Themes](https://wordpress.org/themes/). Anyone is welcome to reopen to continue the discussion as needed.
